sqlmesh/core/test/definition.py

```diff
@@ -337,7 +335,8 @@ def _test_fixture_table(self, name: str) -> exp.Table:
         table = self._fixture_table_cache.get(name)
         if not table:
             table = exp.to_table(name, dialect=self.dialect)
-            table.this.set("this", f"{table.this.this}__fixture__{self._test_id}")
+            table.set("db", exp.to_identifier(self._test_schema))
```

Comment:
Don't we want to somehow encode the original schema name in the test one? Or otherwise include the original schema name in the table name? What happens if the 2 upstream tables share the same table name but are in different schemas?

Reply:
Yes, that's a valid point. We could prepend the schema of each fixture, if any, to its table name. So for example `sushi.foo` would be accessed as `sqlmesh_test_some_id.sushi_foo`. Wdyt about this approach?